controllers/clusteroperator_test.go

```diff
@@ -266,3 +281,46 @@ func TestUpdateCOStatusDegraded(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdateCOStatusAvailable(t *testing.T) {
+	tCases := []struct {
+		name               string
+		expectedConditions []osconfigv1.ClusterOperatorStatusCondition
+		expected           bool
+	}{
+		{
+			name: "Incorrect Condition",
+			expectedConditions: []osconfigv1.ClusterOperatorStatusCondition{
+				setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionFalse, "", ""),
+				setStatusCondition(OperatorDisabled, osconfigv1.ConditionFalse, "", ""),
+			},
+			expected: false,
+		},
+		{
+			name: "Correct Condition",
+			expectedConditions: []osconfigv1.ClusterOperatorStatusCondition{
+				setStatusCondition(osconfigv1.OperatorAvailable, osconfigv1.ConditionTrue, "", ""),
+				setStatusCondition(OperatorDisabled, osconfigv1.ConditionFalse, "", ""),
+			},
+			expected: true,
+		},
+	}
+
+	reconciler := newFakeProvisioningReconciler(setUpSchemeForReconciler(), &osconfigv1.Infrastructure{})
+	co, _ := reconciler.createClusterOperator()
+	reconciler.OSClient = fakeconfigclientset.NewSimpleClientset(co)
+
+	for _, tc := range tCases {
+		reconciler.updateCOStatusAvailable()
+		gotCO, _ := reconciler.OSClient.ConfigV1().ClusterOperators().Get(context.Background(), clusterOperatorName, metav1.GetOptions{})
+
+		for _, expectedCondition := range tc.expectedConditions {
+			ok := v1helpers.IsStatusConditionPresentAndEqual(
+				gotCO.Status.Conditions, expectedCondition.Type, expectedCondition.Status,
+			)
+			if !ok {
+				assert.Regexp(t, tc.expected, ok)
```

Comment:
This seems to be using a regex to compare 2 booleans?

Reply:
Yes. Will fix it.